docs/specs/servers/push/spec.md

```diff
@@ -80,27 +91,51 @@ DELETE <PUSH_SERVER_URL>/clients/<CLIENT_ID>
 
 ### Send Notification
 
+This endpoint is called by the relay.
+
 ```
 POST <PUSH_SERVER_URL>/clients/<CLIENT_ID>
+```
+
+- `CLIENT_ID` - This is used to map to the device token
+
+Headers:
+- `X-Ed25519-Timestamp` - timestamp of the signature
+- `X-Ed25519-Signature` - hex encoded ed25519 signature of `format!("{}.{}.{}", timestamp, body.len(), body)` where `body` is the request body
+
+Request body:
+```typescript
 {
-    "id": "0000-0000-0000-0000"
-    "payload": {
-      "topic": "<topic>",
-      "flags": <flags>,
-      "blob": "<blob>"
-    }
-```
-
-- `CLIENT_ID`: This is used to map to the device token
-- `id`: This is generated by the Relay to prevent duplicate notifications
-- `payload`: A string with 3 values encoded and split with a `:` as a delimiter.
-  - `topic`: This is used by the SDKs to decrypt encrypted payloads on the client side
-  - `flags`: an integer value that uses bitmasks to encode information about the payload for the SDKs, the flags are below
-    ```
-    Encrypted  = 1 << 0
-    Sign       = 1 << 1
-    Auth       = 1 << 2
-    Chat       = 1 << 3
-    Notify     = 1 << 4
-    ```
-  - `blob`: If the payload is encrypted this is just the encrypted blob, when this is cleartext it should be base64 encoded. The blob should follow the format from of the [Notify Message](https://specs.walletconnect.com/2.0/specs/clients/notify/data-structures#notify-message)
+  // Topic of the message. This is used by the SDKs to decrypt encrypted payloads on the client side
+  topic: string,
+
+  // The relay message.
+  message: string,
+
+  // ==== Legacy Fields ====
```

Comment:
Migration milestones:
- Relay and Push Server adopts this new spec
- Wallet clients migrate to fully decrypted notifications
- Once all clients migrated, we can remove Legacy Fields

Reply:
Updated specs to show that Push Server will now generate the cleartext message: [d83fc21](https://github.com/WalletConnect/walletconnect-specs/pull/147/commits/d83fc21af042c411ba08dfa16c6526324314d0f8)

Legacy Fields are now fully deprecated, and updated Push Server versions should not depend on them. Once Push Servers update, we can remove these fields from the relay. We can now handle the migration (forced or not) to decrypted notifications separately.

Reply:
@chris13524 Does it break legacy users? Or does it keep them receiving simple/unencrypted notifications?

Reply:
@jakubuid legacy users will continue receiving simple notifications, unless always_raw is set.

Here are updated migration steps:
- Relay and Push Servers adopt this new spec (in no particular order)
- Once both servers migrated, Relay and Push Servers remove Legacy Fields

Optional:
- All wallet clients migrate to fully decrypted notifications
- Push Servers force always_raw: true, and legacy support removed